src/lib/libdylink.js

```diff
@@ -904,7 +904,9 @@ var LibraryDylink = {
     // We need to set rpath in flags based on the current library's rpath.
     // We can't mutate flags or else if a depends on b and c and b depends on d,
     // then c will be loaded with b's rpath instead of a's.
-    flags = {...flags, rpath: { parentLibPath: libName, paths: metadata.runtimePaths }}
+    var dso = LDSO.loadedLibsByName[libName];
+    var libPath = dso?.path ?? libName;
+    flags = {...flags, rpath: { parentLibPath: libPath, paths: metadata.runtimePaths }}
```

Comment:
Should `parentLibPath` be just the dirname part of the parent?   i.e. is there any point in passing a value if its just a basename?

Reply:
I think it is possible to calculate the dirname here.

For now, the dirname is calculated inside [replaceORIGIN](https://github.com/emscripten-core/emscripten/blob/8bc8fff4397781587b2075c17e02e5984cef74e9/src/lib/libdylink.js#L52-L58) function, and the reason I did that was to isolate the code that depends on `FS` or `PATH` as much as possible.

If we calculate the dirname here, I think it can be something like:

```
    var dso = LDSO.loadedLibsByName[libName];
#if FILESYSTEM
    var libPath = PATH.dirname(dso?.path ?? libName);
#endif
    flags = {...flags, rpath: { parentLibPath: libPath, paths: metadata.runtimePaths }}
```